Do-It/Controller/DueDatePickerTableViewController.swift

```diff
@@ -0,0 +1,155 @@
+//
+
+import UIKit
+
+protocol DueDatePickerTableViewControllerDelegate: AnyObject {
+    func dueDatePickerTableViewController(_ viewController: DueDatePickerTableViewController,
+                                          didSelectTimeRange range: DateInterval?)
+}
+
+class DueDatePickerTableViewController: UITableViewController {
+
+    var timeRangeSelected: DateInterval?
+
+    struct Option {
+
+        let index: Int
+        let name: String
+        let dateRange: DateInterval
+
+        init(row: Int, description: String, range: DateInterval) {
+            index = row
+            name = description
+            dateRange = range
+        }
+    }
+
+    var options: [Option] = []
+
+    weak var delegate: DueDatePickerTableViewControllerDelegate?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        makeTimeOptions()
+    }
+
+    func makeTimeOptions() {
+
+        var components = DateComponents()
+        components.day = 1
+        components.second = -1
+
+        let startOfDay = Calendar.current.startOfDay(for: Date())
+        let endOfDay = Calendar.current.date(byAdding: components, to: startOfDay)!
+        let endOfTomorrow = Calendar.current.date(byAdding: .day, value: 1, to: endOfDay)!
+
+        options.append(Option(row: 0,
+                              description: "Today",
+                              range: DateInterval(start: Date(), end: endOfDay)))
+        options.append(Option(row: 1,
+                              description: "Tomorrow",
+                              range: DateInterval(start: endOfDay, end: endOfTomorrow)))
+        options.append(Option(row: 2,
+                              description: "Within the next Three Days",
+                              range: DateInterval(start: Date(),
+                                                  end: Calendar.current.date(byAdding: .day, value: 3, to: Date())!)))
+        options.append(Option(row: 3,
+                              description: "Within the next Week",
+                              range: DateInterval(start: Date(),
+                                                  end: Calendar.current.date(byAdding: .day, value: 7, to: Date())!)))
+
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return options.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DueDateCell", for: indexPath)
+        cell.textLabel?.text = options[indexPath.item].name
+        return cell
+    }
```

Comment:
We want to support the case where the user has already chosen a date range and is now returning to the screen to change their selection. If the date range of the option at this index path patches the selected range, the checkbox should be assigned here.